client/src/components/output.js

```diff
@@ -101,57 +90,63 @@ class Output extends React.Component {
     };
 
     let debtors = this.state.debtors;
-    if (names.indexOf(name) === -1) {
-      this.addDebtor(debtor, itemAndPrice, names, foodList);
-    } else {
-      this.findDebtor(debtors, name, itemAndPrice, foodList);
+    if ( debtors.length === 0) {
+      this.addDebtor(debtor, itemAndPrice);
+    } else if ( debtors.length > 0 ) { 
+      if ( names.indexOf(name) === -1 ) {
+        this.addDebtor(debtor, itemAndPrice);
+      } else {
+        this.findDebtor(debtors, name, itemAndPrice);
+      }
     }
   }
 
-  addDebtor(debtor, itemAndPrice, names, foodList) {
+  addDebtor(debtor, itemAndPrice) {
     debtor.items.push(itemAndPrice);
     foodList.push(itemAndPrice.itemName); 
     let debtorInfo = this.state.debtors.concat(debtor);
     this.setState({
       debtors: debtorInfo
-    }, () => this.helperSetState(names));
+    }, this.helperSetState);
   }
 
-  findDebtor(debtors, name, itemAndPrice, foodList) {
-    for (let i = 0; i < debtors.length; i++) {
-      if (debtors[i].name === name ) {
+
+  findDebtor(debtors, name, itemAndPrice) {
+    for ( let i = 0; i < debtors.length; i++) {
+      if ( debtors[i].name === name ) {
         foodList.push(itemAndPrice.itemName); 
         debtors[i].items.push(itemAndPrice);
       } 
-      if (foodList.indexOf(itemAndPrice.itemName) !== -1 && debtors[i].name !== name ) {
-        var items = debtors[i].items; 
-        for (var j = 0; j < items.length; j++ ) {
-          if (items[j].itemName === itemAndPrice.itemName ) {
+      if ( foodList.indexOf(itemAndPrice.itemName) !== -1 && debtors[i].name !== name ) {
+        let items = debtors[i].items; 
+        for ( let j = 0; j < items.length; j++ ) {
+          if ( items[j].itemName === itemAndPrice.itemName ) {
             items.splice(j, 1);
           } 
         }
       }
     }
   }
 
-  helperSetState (names) {
+  helperSetState () {
```

Comment:
I need to refactor the line of code. I put the names array in global variable in the way I didn't need to pass in any parameter. but when I refactor it then I will need to pass in the variable to helperSetState method.